openedx/core/djangoapps/dark_lang/middleware.py

```diff
@@ -117,14 +129,104 @@ def _activate_preview_language(self, request):
         Check the user's dark language setting in the session and apply it
         """
         auth_user = request.user.is_authenticated()
-        preview_lang = None
-        if auth_user:
+
+        if request.method == 'POST':
+            self.process_darklang_request(request)
+        elif auth_user:
             # Get the request user's dark lang preference
             preview_lang = get_user_preference(request.user, DARK_LANGUAGE_KEY)
 
-        # User doesn't have a dark lang preference, so just return
-        if not preview_lang:
-            return
+            if preview_lang:
+                # Set the session key to the requested preview lang
+                request.session[LANGUAGE_SESSION_KEY] = preview_lang
+
+    def process_darklang_request(self, request):
+        """
+        Proccess the request to set or clear the DarkLang depending on the incoming request.
+
+        Arguments:
+            request (Request): The Django Request Object
 
-        # Set the session key to the requested preview lang
-        request.session[LANGUAGE_SESSION_KEY] = preview_lang
+        Returns:
+            HttpResponse: View containing the form for setting the preview lang with the status
+                included in the context
+        """
+        context = {
+            'disable_courseware_js': True,
+            'uses_bootstrap': True,
+        }
+        response = None
+        if not DarkLangConfig.current().enabled:
+            register_error_message(request, _('Preview Language is currently disabled'))
+        elif 'set_language' in request.POST:
+            # Set the Preview Language
+            self._set_preview_language(request, context)
+        elif 'reset' in request.POST:
+            # Reset and clear the language preference
+            self._clear_preview_language(request, context)
+
+    def _set_preview_language(self, request, context):
+        """
+        Set the Preview language
+
+        Arguments:
+            request (Request): The incoming Django Request
+            context dict: The basic context for the Response
+
+        Returns:
+            HttpResponse: View containing the form for setting the preview lang with the status
+                included in the context
+        """
+        preview_lang = request.POST.get(LANGUAGE_INPUT_FIELD, '')
+        if not preview_lang.strip():
+            register_error_message(request, _('Language code not provided'))
+        else:
+            # Set the session key to the requested preview lang
+            request.session[LANGUAGE_SESSION_KEY] = preview_lang
+
+            # Make sure that we set the requested preview lang as the dark lang preference for the
+            # user, so that the lang_pref middleware doesn't clobber away the dark lang preview.
+            auth_user = request.user
+            if auth_user:
+                set_user_preference(request.user, DARK_LANGUAGE_KEY, preview_lang)
+
+            register_success_message(
```

Comment:
Do you think it's worth alerting the user if the code they inputted is a code that we can handle (ie: if you type in 'abcdef' it returns 'Language set to language code: abcdef' as opposed to 'abcdef is not a language code in our system', or something like that.